File: RT_IMS_front-end/rt_ims_front-end/src/types/stock.ts
import type { EntityId } from "./auth";

export type StockStatus = "Parking" | "Confirmed" | "Issued" | "Return";

export interface StockItem {
  id: EntityId;
  asnGrnId?: EntityId;
  asnGrnNo: string;
  rfidTagId?: EntityId;
  tagValue?: string;
  lotNo: string;
  itemYarnType: string;
  supplier: string;
  bagQty: number;
  weightPerBag: number;
  totalWeight: number;
  status: StockStatus | string;
  lastMovementAt?: string;
}

export interface StockTransaction {
  id: EntityId;
  referenceNo: string;
  tagNo: string;
  tagValue?: string;
  itemYarnType: string;
  lotNo: string;
  transactionType: "Receive" | "MRR Approved" | "Issue" | "Return" | string;
  quantity: number;
  weight: number;
  createdAt: string;
  createdBy: string;
}

export interface StockSummary {
  parkingStock: number;
  confirmedStock: number;
  issuedStock: number;
  returnStock: number;
  warningCount: number;
}

File: RT_Inventory.Api/RT_Inventory.Api/Models/Entities/Stock.cs
using RT_Inventory.Api.Models.Enums;

namespace RT_Inventory.Api.Models.Entities;

public class Stock
{
    public int Id { get; set; }
    public int AsnGrnId { get; set; }
    public int RfidTagId { get; set; }
    public string LotNo { get; set; } = string.Empty;
    public string ItemYarnType { get; set; } = string.Empty;
    public decimal Weight { get; set; }
    public StockStatus Status { get; set; } = StockStatus.Parking;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public AsnGrn AsnGrn { get; set; } = null!;
    public RfidTag RfidTag { get; set; } = null!;
}
